src/include/timestamp_inline.h

```diff
@@ -173,7 +173,10 @@
           WT_MAX((tw)->durable_start_ts, (ta)->newest_start_durable_ts);       \
         (ta)->newest_stop_durable_ts =                                         \
           WT_MAX((tw)->durable_stop_ts, (ta)->newest_stop_durable_ts);         \
-        (ta)->oldest_start_ts = WT_MIN((tw)->start_ts, (ta)->oldest_start_ts); \
+        if ((tw)->start_ts != WT_TS_NONE) \
+            (ta)->oldest_start_ts = WT_MIN((tw)->start_ts, (ta)->oldest_start_ts); \
+        if ((tw)->start_prepare_ts != WT_TS_NONE)  \
```

Comment:
@quchenhao currently `start_prepare_ts` and `stop_prepare_ts` requires the other PR to be merged first to compile here